app/controllers/friendships_controller.rb

```diff
@@ -4,16 +4,20 @@ def create
     if new_friend == []
       flash[:notice] = 'Email Address not in our system.'
     else
-      already_friend = User.where(id: Friendship.where(user_id: new_friend[0].id).pluck(:friend_id)).where(id: current_user.id)
-      if current_user.id == new_friend[0].id
-        flash[:notice] = "Well that's your email address! We do like that you're trying to be your own friend though :)"
-      elsif already_friend[0].nil?
-        Friendship.create(user_id: current_user.id, friend_id: new_friend[0].id)
-        Friendship.create(user_id: new_friend[0].id, friend_id: current_user.id)
-      else
-        flash[:notice] = "They're already your friend!"
-      end
+      correct_email(new_friend)
     end
     redirect_to '/user/dashboard'
   end
+
+  def correct_email(new_friend)
+    friend = User.where(id: Friendship.where(user_id: new_friend[0].id).pluck(:friend_id)).where(id: current_user.id)
```

Comment:
This active record query seems like it may be redundant, I definitely need a walkthrough for this

Reply:
Let's definitely look at that for the refactor!